### ISSUE TITLE
Can't reinstall bundled plugins

### ISSUE THREAD

**Issue by alexandrebastien** [splitbrain/dokuwiki#1451]
For some reason, if you uninstalled one of the bundled plugin, the manager won’t let you re-install them (the install button is not there for bundled plugins). It is also not possible to reinstall manually since the download button for these plugins is not present on the dokuwiki website. A good workaround would be to make these available for download on the site, until the bug is fixed. This situation affects me particulary because our wiki is hosted on a university server which I cannot access directly. So I can't correct the situation by manually copying the files. I'm using release 2015-08-10a "Detritus".

**Comment by alexandrebastien** [comment]
There's a forum entry with more info here: 
https://bugs.dokuwiki.org/index.php?do=details&task_id=2946

**Comment by Klap-in** [comment]
Does the upgrade plugin re-install it maybe?

Op donderdag 21 januari 2016 heeft alexandrebastien <
notifications@github.com> het volgende geschreven:

**Comment by alexandrebastien** [comment]
No, the upgrade plugin says: "Il y as des erreurs! Vous ne devriez < b > pas < / b > continuer !
" (meaning: there is some errors, you should not continue). And only the stop button is available.

**Comment by splitbrain** [comment]
You can continue with upgrade plugin, even when it says you shouldn't. Bundled plugins do not have individual downloads so they can't be reinstalled via the extension manager.

We had decided that some of the less important bundled plugins should be deinstallable. The question remains if we should rethink that decision and mark all bundled plugins as essential. Opinions?

**Comment by alexandrebastien** [comment]
Forcing the upgrade actually worked for me.

In my opinion, I think it's a good idea to let users uninstall some of the less important bundled plugins, but these should have a download link to make them easier to reinstall.

**Comment by dom-mel** [comment]
When i uninstall a plugin i get the following benefits:
1. Less unimportant things at the config page (in fact the auth plugins take some space)
2. more free disk space

To 1. IMHO this is an important thing for users to get a clean and simple view of DokuWiki. But i guess this can be achieved by disabling a plugin?
To 2. for current development version all bundled plugins are 2.6M. 
But on the other hand I know that some users deleted unused language files to save some space which are only 1.9M in total. So this is a tough one.

IMHO best for the user could be the possibility to uninstall bundled plugins. Since we provide a open and extensible software it should be to the user to remove certain parts in an easy way.
But bundled plugins are somehow important. We should provide an easy way of reinstalling them via Extension Manager.

**Comment by michitux** [comment]
A problem with providing downloads is that more recent bundled plugins might not be compatible with older versions of DokuWiki. Therefore providing a simple download URL for them won't work as then users of older versions will upgrade the plugins and break their installation. For a proper solution we need support for plugin downloads that depend on the DokuWiki version in the plugin repository and the extension manager. I think all in all this would be a good solution that could also be beneficial for other plugins where multiple DokuWiki versions shall be supported.
